genapi/genapi.go

```diff
@@ -241,16 +246,38 @@ func (g *GenAPI) APIMode() {
 	kv["addr"] = g.ListenAddr
 
 	// Once ListenAddr is populated with the final value we can call doSkyAPI
-	g.doSkyAPI()
+	skyapiStopCh := g.doSkyAPI()
+
+	hw := &httpWaiter{
+		ch: make(chan struct{}, 1),
+	}
 
 	srv := &http.Server{
 		Addr:    g.ListenAddr,
-		Handler: h,
+		Handler: hw.handler(h),
 	}
 
-	llog.Info("starting rpc listening", kv)
-	kv["err"] = srv.Serve(tcpKeepAliveListener{ln.(*net.TCPListener)})
-	llog.Fatal("rpc listening failed", kv)
+	sigCh := make(chan os.Signal, 1)
+	//signal.Notify(sigCh, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
lol nooooope.... now I have to go back and figure out how this was working at all.....